### PR TITLE
Initial support for validating extension engines.lens version

### PR DESCRIPTION
**Pull request by jakolehm**
image

### CODE REVIEW

File: src/extensions/extension-discovery.ts [hunk 1]
```diff
@@ -47,6 +49,7 @@ export interface InstalledExtension {
   // Absolute to the symlinked package.json file
   readonly manifestPath: string;
   readonly isBundled: boolean; // defined in project root's package.json
+  readonly isValid: boolean;
```
Comment (jim-docker) [inline src/extensions/extension-discovery.ts]: For readability `isCompatible` would be more accurate?

**Review by jim-docker — DISMISSED** [review]
just noticed this is draft. This is just a stop-gap? Are we planning to manage extensions via the catalog? Anyways, tested, works.

**Comment by jakolehm** [comment]
> just noticed this is draft. This is just a stop-gap? Are we planning to manage extensions via the catalog? Anyways, tested, works.

Marked as draft because tests are still failing.

**Review by nevalla — DISMISSED** [review]
LGTM

**Review by Nachasic — DISMISSED** [review]

**Comment by nevalla** [comment]
Tests are again failing after the rebase.

File: src/extensions/extension-discovery.ts (hunk 1 above)
Comment (Nokel81) [reply src/extensions/extension-discovery.ts]: I would also say that `isValid` and `isCompatible` don't really mean the same thing.

File: src/extensions/extension-discovery.ts [hunk 2]
```diff
@@ -349,20 +353,26 @@ export class ExtensionDiscovery extends Singleton {
...
       const installedManifestPath = this.getInstalledManifestPath(manifest.name);
       const isEnabled = isBundled || ExtensionsStore.getInstance().isEnabled(installedManifestPath);
       const extensionDir = path.dirname(manifestPath);
       const npmPackage = path.join(extensionDir, `${manifest.name}-${manifest.version}.tgz`);
       const absolutePath = (isProduction && await fse.pathExists(npmPackage)) ? npmPackage : extensionDir;
+      let isValid = false;
+
+      if (manifest.engines?.lens) {
+        isValid = semver.satisfies(appSemVer, manifest.engines.lens);
+      }
```
Comment (Nokel81) [inline src/extensions/extension-discovery.ts]:
```suggestion
      const isCompatible = manifest.engines?.lens 
      	? semver.satisfies(appSemVer, manifest.engines.lens)
      	: false;
```

File: src/extensions/extension-installer.ts [hunk 3]
```diff
@@ -25,17 +25,10 @@ import fs from "fs-extra";
...
-  [name: string]: string;
-};
-
-// Type for the package.json file that is written by ExtensionInstaller
-export type PackageJson = {
-  dependencies: Dependencies;
-};
 
 /**
  * Installs dependencies for extensions
```
Comment (Nokel81) [inline src/extensions/extension-installer.ts:34]: I also think that we should check if the extension is not-compatible before trying to install it either.

File: src/extensions/extension-discovery.ts (hunk 1 above)
Comment (jakolehm) [reply src/extensions/extension-discovery.ts]: Much better, fixed.

File: src/extensions/extension-installer.ts (hunk 3 above)
Comment (jakolehm) [reply src/extensions/extension-installer.ts:34]: Let's do that in a separate PR.

Comment (Nokel81) [reply src/extensions/extension-installer.ts:34]: Sounds good

**Review by Nokel81 — DISMISSED** [review]

File: src/renderer/components/+extensions/installed-extensions.tsx [hunk 4]
```diff
@@ -102,29 +106,34 @@ export const InstalledExtensions = observer(({ extensions, uninstall, enable, di
...
-                </MenuItem>
-              ) : (
-                <MenuItem
-                  disabled={isUninstalling}
-                  onClick={() => enable(id)}
-                >
-                  <Icon material="check_circle"/>
-                  <span className="title" aria-disabled={isUninstalling}>Enable</span>
-                </MenuItem>
+              { isCompatible && (
```
Comment (jim-docker) [inline src/renderer/components/+extensions/installed-extensions.tsx:115]: Might be nice to one day have an "Install latest" or "find compatible" option when the the extension is not compatible, but that really means knowing if there is a compatible version available...

**Review by jim-docker — DISMISSED** [review]

File: src/renderer/components/+extensions/installed-extensions.tsx (hunk 4 above)
Comment (jakolehm) [reply src/renderer/components/+extensions/installed-extensions.tsx:115]: There has been discussions about moving extensions to catalog. Catalog could show list of all known extensions with actions to install/uninstall/update.

**Review by Nachasic — DISMISSED** [review]

File: src/common/vars.ts [hunk 5]
```diff
@@ -66,4 +67,6 @@ export const apiKubePrefix = "/api-kube" as string; // k8s cluster apis
 export const issuesTrackerUrl = "https://github.com/lensapp/lens/issues" as string;
 export const slackUrl = "https://join.slack.com/t/k8slens/shared_invite/enQtOTc5NjAyNjYyOTk4LWU1NDQ0ZGFkOWJkNTRhYTc2YjVmZDdkM2FkNGM5MjhiYTRhMDU2NDQ1MzIyMDA4ZGZlNmExOTc0N2JmY2M3ZGI" as string;
 export const supportUrl = "https://docs.k8slens.dev/latest/support/" as string;
+
+export const appSemVer = new SemVer(packageInfo.version);
```
Comment (Nokel81) [inline src/common/vars.ts:71]: nit: maybe `appVersion`?

**Review by Nokel81 — APPROVED** [review]

**Pull request merged**